### PR TITLE
Add unit tests and types file

### PR DESCRIPTION
**Pull request by danroc**
Thir PR adds some unit tests back. They do not cover 100% of the code but should cover the principal areas of it.

### CODE REVIEW

File: src/snap-keyring.ts [hunk 1]
```diff
@@ -0,0 +1,355 @@
...
+   * @param state - Serialized keyring state.
+   */
+  async deserialize(state: KeyringState | undefined): Promise<void> {
+    if (state === undefined) {
+      return;
+    }
+
+    assert(state, KeyringStateStruct);
+    this.#addressToAccount = state.addressToAccount;
+    this.#addressToSnapId = state.addressToSnapId;
```
Comment (montelaidev) [inline src/snap-keyring.ts:141]: we should set #pendingRequests as well

File: src/snap-keyring.ts [hunk 2]
```diff
@@ -0,0 +1,355 @@
...
+
+  /**
+   * Serialize the keyring state.
+   *
+   * @returns Serialized keyring state.
+   */
+  async serialize(): Promise<KeyringState> {
+    return {
+      addressToAccount: this.#addressToAccount,
+      addressToSnapId: this.#addressToSnapId,
```
Comment (montelaidev) [inline src/snap-keyring.ts:125]: serialize pendingRequests

File: src/snap-keyring.ts (hunk 1 above)
Comment (danroc) [reply src/snap-keyring.ts:141]: Good point, but it may not be possible due to the promises. Maybe we can leave it for a different PR?

**Review by gantunesr — APPROVED** [review]

File: src/snap-keyring.ts (hunk 2 above)
Comment (danroc) [reply src/snap-keyring.ts:125]: Same as below.

**Pull request merged**